internal/exec/shell_utils.go

```diff
@@ -263,21 +278,24 @@ func execTerraformShellCommand(
 		}
 	}
 
+	// Merge env vars, ensuring componentEnvList takes precedence
+	mergedEnv := mergeEnvVars(os.Environ(), componentEnvList)
+
 	u.LogDebug(atmosConfig, "\nStarting a new interactive shell where you can execute all native Terraform commands (type 'exit' to go back)")
 	u.LogDebug(atmosConfig, fmt.Sprintf("Component: %s\n", component))
 	u.LogDebug(atmosConfig, fmt.Sprintf("Stack: %s\n", stack))
 	u.LogDebug(atmosConfig, fmt.Sprintf("Working directory: %s\n", workingDir))
 	u.LogDebug(atmosConfig, fmt.Sprintf("Terraform workspace: %s\n", workspaceName))
 	u.LogDebug(atmosConfig, "\nSetting the ENV vars in the shell:\n")
-	for _, v := range componentEnvList {
+	for _, v := range mergedEnv {
 		u.LogDebug(atmosConfig, v)
 	}
```

Comment:
No need for a separate loop. Maybe just move the log debug statements to `mergeEnvVars` where we already iterate over them.